test/interoperability/interop_test_suite.py

```diff
@@ -0,0 +1,537 @@
+import clients
+import itertools
+import inspect
+import datetime
+import uuid
+import collections
+import time
+import logging
+
+from typing import Dict, Any, List
+
+logging.basicConfig(level=logging.INFO)
+LOG = logging.getLogger(__name__)
+
+DATE_FORMAT = "%Y-%m-%dT%H:%M:%SZ"
+
+VERTICES = [
+    {"lng": 130.6205, "lat": -23.6558},
+    {"lng": 130.6301, "lat": -23.6898},
+    {"lng": 130.6700, "lat": -23.6709},
+    {"lng": 130.6466, "lat": -23.6407},
+]
+
+GEO_POLYGON_STRING = ",".join("{},{}".format(x["lat"], x["lng"]) for x in VERTICES)
+
+SHORT_WAIT_SEC = 5
+
+TestContext = collections.namedtuple("TestContext", ["type", "key"])
+
+class InterOpTestSuite:
+    def __init__(self, dss_clients: Dict[str, clients.DSSClient]):
+        self.dss_clients = dss_clients
+
+    def startTest(self):
+        round = 1
+        for dss_permutation in itertools.permutations(self.dss_clients):
+            primary_dss = dss_permutation[0]
+            all_other_dss = list(dss_permutation[1:])
+            ts = TestSteps()
+            LOG.info(f"Round {round}")
+            for name, test_step in self._getTests().items():
+                try:
+                    test_step(
+                        ts, self.dss_clients, primary_dss, all_other_dss=all_other_dss
+                    )
+                    LOG.info(f"{name} Passed with {primary_dss} as primary DSS")
+                except AssertionError as e:
+                    docstring = inspect.cleandoc(inspect.getdoc(test_step))
+                    msg = (
+                        f"Failed {name} with {primary_dss} as primary DSS\n"
+                        f"\tTest Purpose: {docstring}\n"
+                        f"\tFailure Message: {e}"
+                    )
+                    LOG.error(msg)
+                    LOG.debug(f"Cleaning up round {round}")
+                    ts.cleanUp(self.dss_clients, primary_dss)
+                    return (primary_dss, msg)
+
+            LOG.debug(f"Cleaning up round {round}")
+            ts.cleanUp(self.dss_clients, primary_dss)
+            round += 1
+
+    def _getTests(self) -> Dict[str, object]:
+        methods = inspect.getmembers(TestSteps, predicate=inspect.isfunction)
+        result: Dict[str, object] = {}
+        ordered_result = collections.OrderedDict()
+        for name, obj in methods:
+            if name.startswith("testStep"):
+                result[name] = obj
+        # sort by name
+        for i in range(1, len(result) + 1):
```

Comment:
So this would end up sorting like this 
testStep1
testStep10
testStep11
.....
testStep2
.....
Since I know that it follows the numbering scheme, I went with this approach. I can write a lambda to make it sort numerically

Reply:
Ah, right.  If leaving as-is, let's at least comment on TestSteps that only methods following the convention testStep%d will be run as test steps.